components/Footer/Form/Form.tsx

```diff
@@ -0,0 +1,207 @@
+import { AiOutlineCheck } from 'react-icons/ai';
+import { useEffect, useState } from 'react';
+
+import { SmallParagraph } from 'components/atoms/typography.styles';
+import {
+  Blue050, Green050, Green100, Grey010,
+} from 'components/styling/colors';
+import PrimaryCTAButton from 'components/atoms/PrimaryCTAButton';
+import {
+  StyledForm,
+  FormFields,
+  ButtonContainer,
+  CheckBox,
+  CheckBoxErrorContainer,
+  CheckBoxWrapper,
+  CheckMark,
+  FormBottom,
+} from 'components/Footer/Form/Form.styles';
+import TextInput from 'components/Footer/Form/molecules/TextInput';
+import MessageInput from 'components/Footer/Form/molecules/MessageInput';
+import ErrorMessage from 'components/Footer/Form/atoms/ErrorMessage';
+import ServiceSelector from 'components/Footer/Form/molecules/ServiceSelector';
+import labelTexts from 'components/Footer/Form/content/labelTexts';
+
+interface FormStateType {
+  hasErrors: boolean;
+  data: {
+    name: string,
+    email: string;
+    phone: string;
+    selectedService: string;
+    message: string;
+    dataHandlingCheckBox: boolean;
+  };
+  errors: {
+    name: string,
+    email: string;
+    message: string;
+    dataHandlingCheckBox: string;
+  };
+  touched: Set<string>;
+}
+
+const emailRegExp = /[^@]+@[^@]+\.[^@]{2,}/;
+
+const Form = () => {
+  const [checked, setIsChecked] = useState(false);
+
+  const [formState, setFormState] = useState<FormStateType>({
+    hasErrors: true,
+    data: {
+      name: '',
+      email: '',
+      phone: '',
+      selectedService: '',
+      message: '',
+      dataHandlingCheckBox: false,
+    },
+    errors: {
+      name: '',
+      email: '',
+      message: '',
+      dataHandlingCheckBox: '',
+    },
+    touched: new Set<string>(),
+  });
+
+  const updateData = (key: string, value: boolean | string) => {
+    const touchedCopy = formState.touched;
+    touchedCopy.add(key);
+
+    setFormState({
+      ...formState,
+      data: { ...formState.data, [key]: value },
+      touched: touchedCopy,
+    });
+  };
+
+  const toggleIsChecked = () => {
+    updateData('dataHandlingCheckBox', !checked);
+    setIsChecked(!checked);
+  };
+
+  const isTouched = (key: string, stateToValidate: FormStateType) =>
+    stateToValidate.touched.has(key);
+
+  const carryOutValidations = (stateToValidate: FormStateType = formState) => {
+    const errors = { ...stateToValidate.errors };
```

Comment:
I don't really understand this part. What is the reason behind spreading the old state, and then recalculating for every field ?

Wouldn't it make sense just to create
`const errors = {
  name: '',
  email: '',
  ...
}`
and then populate it ?